### PR TITLE
Prevent duplicate package reporting

### LINKED ISSUE #1195: Avoid adding SDK info twice to event
Transactions can end up double reporting packages:

image

### PR DESCRIPTION
**Pull request by josh-degraw**
Fixes #1195

### CODE REVIEW

**Review by bruno-garcia — APPROVED** [review]
LGTM

**Comment by bruno-garcia** [comment]
> 
D:\a\sentry-dotnet\sentry-dotnet\src\Sentry\Package.cs(63,29): error CS1591: Missing XML comment for publicly visible type or member 'Package.GetHashCode()' [D:\a\sentry-dotnet\sentry-dotnet\src\Sentry\Sentry.csproj]

Little gotcha: we only warn missing XML docs on `Release` builds, so we can dev in peace in Debug mode and only clean that up before the final push. `./build.sh` does that check if u want to do a test before pushing

**Comment by lucas-zimerman** [comment]
Since Danger doesn't show the message here for PRs outisde of the main fork.
```
Please consider adding a changelog entry for the next release.

Instructions and example for changelog
Please add an entry to `CHANGELOG.md` to the "Unreleased" section. Make sure the entry includes this PR's number.

Example:
Prevent duplicate package reporting ([#1197](https://github.com/getsentry/sentry-dotnet/pull/1197))
If none of the above apply, you can opt-out of this check by adding `#skip-changelog` to the PR description.
```

File: src/Sentry/Package.cs
Comment (lucas-zimerman) [inline src/Sentry/Package.cs:67]: Why 397?

Comment (bruno-garcia) [reply src/Sentry/Package.cs:67]: prime number

**Comment by bruno-garcia** [comment]
> Since Danger doesn't show the message here for PRs outisde of the main fork.
> 
> ```
> Please consider adding a changelog entry for the next release.
> 
> Instructions and example for changelog
> Please add an entry to `CHANGELOG.md` to the "Unreleased" section. Make sure the entry includes this PR's number.
> 
> Example:
> Prevent duplicate package reporting ([#1197](https://github.com/getsentry/sentry-dotnet/pull/1197))
> If none of the above apply, you can opt-out of this check by adding `#skip-changelog` to the PR description.
> ```

You have write access to the repo, you can change your remote to this and remove ur fork

File: src/Sentry/Package.cs
Comment (josh-degraw) [reply src/Sentry/Package.cs:67]: I never actually knew why, that's just auto-generated for me haha. Good to know why though!

**Comment by lucas-zimerman** [comment]
Test failed 
```
  Sentry.Log4Net.Tests -> /Users/runner/work/sentry-dotnet/sentry-dotnet/test/Sentry.Log4Net.Tests/bin/Release/netcoreapp2.1/Sentry.Log4Net.Tests.dll
  Sentry.AspNetCore.Grpc.Tests -> /Users/runner/work/sentry-dotnet/sentry-dotnet/test/Sentry.AspNetCore.Grpc.Tests/bin/Release/netcoreapp3.1/Sentry.AspNetCore.Grpc.Tests.dll
  Sentry.EntityFramework.Tests -> /Users/runner/work/sentry-dotnet/sentry-dotnet/test/Sentry.EntityFramework.Tests/bin/Release/netcoreapp3.1/Sentry.EntityFramework.Tests.dll
[xUnit.net 00:00:24.40]     Sentry.Tests.Protocol.SdkVersionTests.SerializeObject_IgnoresDuplicatePackages [FAIL]
Error: Sentry.Tests.Protocol.SdkVersionTests.SerializeObject_IgnoresDuplicatePackages: Assert.Equal() Failure
  Failed Sentry.Tests.Protocol.SdkVersionTests.SerializeObject_IgnoresDuplicatePackages [< 1 ms]
  Error Message:
   Assert.Equal() Failure
                                 ↓ (pos 34)
Expected: ···"name":"nuget:Sentry.Extensions.Logging","version":"3.9.2"},{···
Actual:   ···"name":"nuget:Sentry","version":"3.9.2"},{"name":"nuget:Sentr···
                                 ↑ (pos 34)
  Stack Trace:
     at Sentry.Tests.Protocol.SdkVersionTests.SerializeObject_IgnoresDuplicatePackages() in /Users/runner/work/sentry-dotnet/sentry-dotnet/test/Sentry.Tests/Protocol/SdkVersionTests.cs:line 74
Test run for /Users/runner/work/sentry-dotnet/sentry-dotnet/test/Sentry.EntityFramework.Tests/bin/Release/netcoreapp3.1/Sentry.EntityFramework.Tests.dll (.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.11.0
Copyright (c) Microsoft Corporation.  All rights reserved.
```

File: test/Sentry.Tests/Protocol/SdkVersionTests.cs
Comment (lucas-zimerman) [inline test/Sentry.Tests/Protocol/SdkVersionTests.cs:96]:
```suggestion
            Assert.Equal(expected, actual);
        }

        private static string TrimJson(string json)
        {
            return Regex.Replace(json, @"\s", "");
        }
```

File: test/Sentry.Tests/Protocol/SdkVersionTests.cs
Comment (lucas-zimerman) [inline test/Sentry.Tests/Protocol/SdkVersionTests.cs:71]: Not sure if I follow, but shouldn't you insert a duplicated package into sdkVersion?
```suggestion
            sdkVersion.AddPackage("Bar", "Beta");
            sdkVersion.AddPackage("Bar", "Beta");
```

**Review by lucas-zimerman — APPROVED** [review]
Other than the mentioned topics, it looks ok for me.

File: test/Sentry.Tests/Protocol/SdkVersionTests.cs
Comment (josh-degraw) [reply test/Sentry.Tests/Protocol/SdkVersionTests.cs:71]: 🤦🏻‍♂️ Yes... I somehow removed that when cleaning this up. Good catch lol

File: CHANGELOG.md
Comment (bruno-garcia) [inline CHANGELOG.md:4]: Ideally this goes under the `### Fixes`

**Review by bruno-garcia — APPROVED** [review]
Small adjustment on the changelog but other than that feel free to merge

**Comment by bruno-garcia** [comment]
@josh-degraw feel free to merge ur own PRs once they are approved

**Comment by josh-degraw** [comment]
> @josh-degraw feel free to merge ur own PRs once they are approved

Sounds good, I was actually just about to ask you that haha.

**Pull request merged**
